internal/matchers/expectation.go

```diff
@@ -185,10 +187,11 @@ func (e *Expectation) NotToContain(expected interface{}) {
 		expectedElem := expectedValue.Index(i).Interface()
 
 		for j := 0; j < actualValue.Len(); j++ {
-			if reflect.DeepEqual(actualValue.Index(j).Interface(), expectedElem) {
-				e.fail(fmt.Sprintf("Expected\n\t%v\nnot to contain\n\t%v", e.actual, expectedElem))
-				return
+			if !reflect.DeepEqual(actualValue.Index(j).Interface(), expectedElem) {
+				continue
 			}
+			e.fail(fmt.Sprintf("Expected\n\t%v\nnot to contain\n\t%v", e.actual, expectedElem))
+			return
```

Comment:
I dislike this change. I think that it violates https://go.dev/wiki/CodeReviewComments#indent-error-flow:

> Try to keep the normal code path at a minimal indentation, and indent the error handling, dealing with it first.

Can we ignore it with `// nolint:revive // We want to keep the error handling indented.